src/Uno.UI.RemoteControl.VS/Helpers/DTEHelper.cs

```diff
@@ -3,18 +3,93 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using EnvDTE;
 using EnvDTE80;
+using Microsoft.VisualStudio.ProjectSystem;
 using Microsoft.VisualStudio.Shell;
 
 namespace Uno.UI.RemoteControl.VS.Helpers;
 
 internal static class DTEHelper
 {
+	internal const string FolderKind = "{66A26720-8FB5-11D2-AA7E-00C04F688DDE}";
+
 	public static int GetMSBuildOutputVerbosity(this DTE2 dte)
 	{
 		ThreadHelper.ThrowIfNotOnUIThread();
 		var properties = dte?.Properties["Environment", "ProjectsAndSolution"];
 		var logOutput = properties?.Item("MSBuildOutputVerbosity").Value is int log ? log : 0;
 		return logOutput;
 	}
+
+	public static async Task<IEnumerable<Project>> GetProjectsAsync(this DTE dte)
+	{
+		await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+		var solutionProjectItems = dte.Solution.Projects;
+
+		if (solutionProjectItems != null)
+		{
+			return EnumerateProjects(solutionProjectItems);
+		}
+		else
+		{
+			return Array.Empty<Project>();
+		}
+	}
+
+	private static IEnumerable<Project> EnumerateProjects(Projects vsSolution)
+	{
+		ThreadHelper.ThrowIfNotOnUIThread();
+
+		foreach (var project in vsSolution.OfType<Project>())
+		{
+			if (project.Kind == FolderKind /* Folder */)
+			{
+				foreach (var subProject in EnumSubProjects(project))
+				{
+					yield return subProject;
+				}
```

Comment:
I would think that if you have a folder you want to end up recursing so that you have projects in nested folders

Reply:
That's what EnumSubProjects does, afaik. (it's been like this for a while, so it's pretty stable :))